zebra-chain/src/block.rs

```diff
@@ -105,3 +133,18 @@ impl TrustedPreallocate for Hash {
         ((MAX_PROTOCOL_MESSAGE_LEN - 1) as u64) / BLOCK_HASH_SIZE
     }
 }
+
+use thiserror::Error;
+
+/// Errors that can occur when checking Block consensus rules.
+///
+/// Each error variant corresponds to a consensus rule, so enumerating
+/// all possible verification failures enumerates the consensus rules we
+/// implement, and ensures that we don't reject blocks or transactions
+/// for a non-enumerated reason.
```

Comment:
This docstring is not as accurate when moved out of `zebra-consensus`

Reply:
I removed that part from the `zebra-chain/block/errors.rs` in https://github.com/ZcashFoundation/zebra/pull/2354/commits/3c5a52192b366b6627c0205d44f2ecff816ca2b5 as it is not accurate.

However the one in [zebra-consensus](https://github.com/ZcashFoundation/zebra/blob/main/zebra-consensus/src/error.rs#L3-L6) is still valid, as the error from `zebra-chain` is casted to [WrongTransactionConsensusBranchId](https://github.com/ZcashFoundation/zebra/blob/main/zebra-consensus/src/error.rs#L157)